core/src/test/java/org/pentaho/reporting/platform/plugin/JobManagerTest.java

```diff
@@ -93,37 +98,42 @@ public class JobManagerTest {
 
     // return mock executor for any call to it's bean name.
     when( objFactory.objectDefined( anyString() ) ).thenReturn( true );
-    when( objFactory.objectDefined( any( Class.class ) ) ).thenReturn( true );
-    when( objFactory.get( any( Class.class ), eq( PentahoAsyncExecutor.BEAN_NAME ), any( IPentahoSession.class ) ) )
+    when( objFactory.get( Mockito.<Class>any(), eq( PentahoAsyncExecutor.BEAN_NAME ), Mockito.<IPentahoSession>any() ) )
       .thenReturn( executor );
 
-    when( objFactory.get( any( Class.class ), eq( "ISchedulingDirectoryStrategy" ), any( IPentahoSession.class ) ) )
+    when( objFactory.get( Mockito.<Class>any(), eq( "ISchedulingDirectoryStrategy" ), Mockito.<IPentahoSession>any() ) )
       .thenReturn( schedulingDirectoryStrategy );
 
     final IUnifiedRepository repository = mock( IUnifiedRepository.class );
-    when( objFactory.get( any( Class.class ), eq( "IUnifiedRepository" ), any( IPentahoSession.class ) ) )
+    when( objFactory.get( Mockito.<Class>any(), eq( "IUnifiedRepository" ), Mockito.<IPentahoSession>any() ) )
       .thenReturn( repository );
 
-    when( objFactory.get( any( Class.class ), eq( "IJobIdGenerator" ), any( IPentahoSession.class ) ) )
+    when( objFactory.get( Mockito.<Class>any(), eq( "IJobIdGenerator" ), Mockito.<IPentahoSession>any() ) )
       .thenReturn( new JobIdGenerator() );
 
     PentahoSystem.registerObjectFactory( objFactory );
   }
 
+  @After
+  public void afterEach() {
+    pentahoSessionHolderMockedStatic.close();
+  }
+
   @AfterClass
   public static void tearDown() throws Exception {
     PentahoSystem.shutdown();
   }
 
   @Before
   public void before() {
+    pentahoSessionHolderMockedStatic = mockStatic( PentahoSessionHolder.class );
```

Comment:
We need to have a good reason for initializing this here instead of in a try-with-resources block in the test method.